ros2topic/test/test_echo_pub.py

```diff
@@ -311,6 +311,57 @@ def publish_message():
                     self.node.destroy_timer(publish_timer)
                     self.node.destroy_publisher(publisher)
 
+    @launch_testing.markers.retry_on_failure(times=1)
+    def test_echo_filter(self, launch_service, proc_info, proc_output):
+        params = [
+            ('/clitest/topic/echo_filter_all_pass', "m.data=='hello'", True),
+            ('/clitest/topic/echo_filter_all_filtered', "m.data=='success'", False),
+
+        ]
+        for topic, filter_expr, has_output in params:
+            with self.subTest(topic=topic, filter_expr=filter_expr, print_count=10):
+                # Check for inconsistent arguments
+                publisher = self.node.create_publisher(String, topic, 10)
+                assert publisher
+
+                def publish_message():
+                    publisher.publish(String(data='hello'))
+
+                publish_timer = self.node.create_timer(0.5, publish_message)
+
+                try:
+                    command_action = ExecuteProcess(
+                        cmd=(['ros2', 'topic', 'echo'] +
+                             ['--filter', filter_expr] +
+                             [topic, 'std_msgs/String']),
+                        additional_env={
+                            'PYTHONUNBUFFERED': '1'
+                        },
+                        output='screen'
+                    )
+                    with launch_testing.tools.launch_process(
+                        launch_service, command_action, proc_info, proc_output,
+                        output_filter=launch_testing_ros.tools.basic_output_filter(
+                            filtered_rmw_implementation=get_rmw_implementation_identifier()
+                        )
+                    ) as command:
+                        # The future won't complete - we will hit the timeout
+                        self.executor.spin_until_future_complete(
+                            rclpy.task.Future(), timeout_sec=5
+                        )
+                    command.wait_for_shutdown(timeout=10)
+                    # Check results
+
```

Comment:
do we need this line?

Reply:
I'm not quite as familiar with the launch testing, but the existing tests used this format as the echo verb never times out on its own so the inner check on spin until future complete is actually what kills the process. Launch wait for shutdown is just to make sure the actual ros processes are cleaned up, I believe

Reply:
sorry i wasn't clear. it is just a nitpick, do we need extra line 354?

Reply:
Oh, not at all. If anything I'd flip 353 and 354. I haven't made many PRs to the ROS codebase yet, so I appreciate the nitpicks.